### PR TITLE
[DEVOPS-61] Add terraform to bootstrap cache-s3 bucket

### PR DESCRIPTION
**Pull request by domenkozar**
Required for https://github.com/input-output-hk/cardano-sl/pull/2166

### CODE REVIEW

File: terraform/appveyor-s3-cache/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+And then modify appveyor.yml with the output values:
+
+environment:
+  global:
+    AWS_ACCESS_KEY_ID:
+      secure: BSNfEghh/xxx=
+    AWS_SECRET_ACCESS_KEY
+      secure: BSNfEghh/xxx=
+    BUCKET: XXX
```
Comment (lehins) [inline terraform/appveyor-s3-cache/README.md]: Same as in https://github.com/input-output-hk/cardano-sl/pull/2166/files#r158594166 variable name is `S3_BUCKET`

**Comment by lehins** [comment]
@domenkozar This looks great!

File: terraform/appveyor-s3-cache/main.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+provider "aws" {
+  region = "us-west-1"
+}
+
+terraform {
+  backend "s3" {
+    bucket  = "iohk-ops-tfstate"
```
Comment (deepfire) [inline terraform/appveyor-s3-cache/main.tf]: @domenkozar, is this snippet with the TF state S3 bucket intended to be copied to all TF deployments, or maybe we could separate it into a separate module, outside of the `appveyor-s3-cache`-specific part?

Comment (domenkozar) [reply terraform/appveyor-s3-cache/main.tf]: Yes, that's going to be shared between deployments. We can do that later on, once we expand tf usage

Comment (lehins) [reply terraform/appveyor-s3-cache/main.tf]: @domenkozar @deepfire It is common to use the same bucket for all deployments done with TF, although having a __separate__ TF state file is a good practice. So, I'd recommend for each logical deployment to have it's own snippet like that, granted with the same bucket name. Reasoning for this is fairly straightforward, you don't want to have one global state for your whole infrastructure. Of course, this is just a suggestion. :)

Comment (domenkozar) [reply terraform/appveyor-s3-cache/main.tf]: @lehins thanks!

**Pull request merged**